### ISSUE TITLE
Latest nvidia driver doesn't compile

### ISSUE THREAD

**Issue by legendofmiracles** [NixOS/nixpkgs#129233]
**Describe the bug**
With the latest kernel and the latest non-beta nvidia driver, the drivers don't compile:
```
error: builder for '/nix/store/wv00lqc6ynx5bhhv31vaixaz867a9vfb-nvidia-x11-460.73.01-5.13.drv' failed with exit code 2;
       last 10 log lines:
       > make[3]: *** Waiting for unfinished jobs....
       > /build/kernel/nvidia-drm/nvidia-drm-modeset.c: In function '__will_generate_flip_event':
       > /build/kernel/nvidia-drm/nvidia-drm-modeset.c:96:23: warning: unused variable 'primary_plane' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
       >    96 |     struct drm_plane *primary_plane = crtc->primary;
       >       |                       ^~~~~~~~~~~~~
       > make[2]: *** [/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/source/Makefile:1847: /build/kernel] Error 2
       > make[2]: Leaving directory '/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/build'
       > make[1]: *** [Makefile:215: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/0l90igcsgvmq2z69ia6n4dalvln2h81w-linux-5.13-dev/lib/modules/5.13.0/source'
       > make: *** [Makefile:80: modules] Error 2
       For full logs, run 'nix log /nix/store/wv00lqc6ynx5bhhv31vaixaz867a9vfb-nvidia-x11-460.73.01-5.13.drv'.
error: 1 dependencies of derivation '/nix/store/gld36fh5sc69s3lw8rnpm701pbsxx67g-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5qnd3i9k96nkxljjs6hpiq100k9ss1m6-linux-5.13-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/11y6fqa7ygg9bbr9mf2rq61ciq6dyzk1-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l29n869335p9xfyvcaqbjgqwc8pkcwl5-nixos-system-dracula-21.11.20210703.09c38c2.drv' failed to build
error: getting status of '/tmp/nixos-rebuild.v0uAIA/result': No such file or directory
```

**To Reproduce**
1. use unstable
2. use the latest linux kernel
3. add nvidia as a drivers

**Expected behavior**
It should compile.

**Notify maintainers**
Latest person who touched it: @ronthecookie 

**Metadata**
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.12, NixOS, 21.11.20210703.09c38c2 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `no` # also happens with it on, it has to be off for other reasons
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `""`
 - channels(nix): `""`
 - nixpkgs: `/nix/store/zy56rp0mfp7lx2xqcfzqcf9z82vph9pd-wsfwxm10shvi4yr0ll8p4v1dp7l7wqfl-source`
```

**Comment by DerDoni** [comment]
I can reproduce this issue, albeit I am using stable channel with the latest kernel.

**Comment by totoroot** [comment]
Same here, on NixOS 21.05...

**Comment by figsoda** [comment]
https://forums.developer.nvidia.com/t/460-67-incompatible-with-kernel-5-13-at-the-moment/177543

**Comment by legendofmiracles** [comment]
tl;dr for everyone facing this:
```nix
boot.kernelPackages = pkgs.linuxPackages_5_12;
```
should serve as a temporary workaround

**Comment by rkitover** [comment]
I think we should implement some kind of kernel version lock for the nvidia driver to prevent such occurrences in the future.

**Comment by figsoda** [comment]
@rkitover https://github.com/NixOS/nixpkgs/issues/122606

**Comment by andrevmatos** [comment]
https://github.com/NixOS/nixpkgs/pull/128785#issuecomment-873219393
